int16 arrow0
int16 arrow1